### PR TITLE
chore: add import and add missing types in fast-element

### PR DESCRIPTION
**Pull request by radium-v**
# Description
* Enables the `importsNotUsedAsValues` tsconfig option for `fast-element`
* Adds missing types
* Organizes imports and converts type-only imports to use `import type`
* Adds the `ElementsFilter` type as an export

## Motivation & context
This is part of a series of PRs needed to allow us to upgrade the API Extractor (and by extension Typescript). The `api-report.md` for `fast-element` doesn't have any `import()` statements but this is part of maintaining the overall consistency with such a large set of changes. 

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-element/src/__test__/helpers.ts [hunk 1]
```diff
@@ -11,6 +11,6 @@ export function toHTML(node: Node, preserveCommentMarkup = false): string {
 }
 
 let counter = 0;
-export function uniqueElementName() {
+export function uniqueElementName(): string {
```
Comment (nicholasrice) [inline packages/web-components/fast-element/src/__test__/helpers.ts:14]: Does this mean we can't infer *any* types?

Comment (radium-v) [reply packages/web-components/fast-element/src/__test__/helpers.ts:14]: No, this is just fixing an eslint error.

File: packages/web-components/fast-element/src/observation/observable.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { DOM } from "../dom";
-import { Notifier, PropertyChangeNotifier, SubscriberSet, Subscriber } from "./notifier";
+import { PropertyChangeNotifier, SubscriberSet } from "./notifier";
+import type { Notifier, Subscriber } from "./notifier";
 
-const volatileRegex = /(\:|\&\&|\|\||if)/;
+const volatileRegex = /(:|&&|\|\||if)/;
```
Comment (EisenbergEffect) [inline packages/web-components/fast-element/src/observation/observable.ts:5]: Thanks!

**Review by EisenbergEffect — APPROVED** [review]
Somehow I totally missed the `import type` feature being added to TypeScript. Nice work @radium-v !

**Review by chrisdholt — APPROVED** [review]

**Comment by radium-v** [comment]
The original title was intentional since [importing types](https://www.typescriptlang.org/docs/handbook/modules.html#importing-types) is the mechanism that this PR was addressing.

**Pull request merged**